Core/Src/bmbUtils.c

```diff
@@ -197,3 +226,46 @@ void insertionSort(Brick_S *arr, int numBricks)
     arr[sortedIdx + 1] = temp;
   }
 }
+
+/*!
+  @brief   Called on a failed trransaction. Partially fills the leaky bucket
+  @param   bucket - The leaky bucket struct to update
+*/
+void updateLeakyBucketFail(LeakyBucket_S* bucket)
+{
+    int32_t bucketFillRemaining = bucket->fillThreshold - bucket->fillLevel;
+    if (bucketFillRemaining <= bucket->failureFillCount)
```

Comment:
Slightly confused by this code. Is seems like if the fill level is ever more than half of the threshold it will fill it all of the way

Reply:
I get it now, making sure the fill level does not go above the threshold. Given I was confused it might need some comments. updateLeakyBucketSuccess has good comments

Reply:
Sounds good, updated